library/network_connections.py

```diff
@@ -332,9 +332,11 @@ def ifcfg_create(
         if connection["type"] == "ethernet":
             ifcfg["TYPE"] = "Ethernet"
             ifcfg["HWADDR"] = connection["mac"]
+            ifcfg["MATCH_PATH"] = connection["pci"]
         elif connection["type"] == "infiniband":
             ifcfg["TYPE"] = "InfiniBand"
             ifcfg["HWADDR"] = connection["mac"]
+            ifcfg["MATCH_PATH"] = connection["pci"]
```

Comment:
initscripts' network-scripts don't support `MATCH_PATH`. That is a NetworkManager extension (like many other options).

Maybe we should have a policy what that means for `provider=initscripts`. Does it make sense to run `provider=initscripts` and write ifcfg files that only work if you afterwards use them with NetworkManager? To me that does not seem useful. I think the role should error out if somebody tries to do such a thing... like, we already do in other cases.

Reply:
right now I know how to tell if a property is supported by initscripts: 
- check if `/etc/sysconfig/network-scripts/ifdown*` and `/etc/sysconfig/network-scripts/ifup*` contain such property

Reply:
@thom311 ,just to confirm, I know we may not assign the Mac address randomly in Network Role, otherwise we have the error like
```
"stderr_lines": ["[001] <error> #0, state:None persistent_state:present, 'eth0': profile specifies mac '00:00:5e:00:53:5d' but no such interface exists"
``` 
but seems like we can random assign the path(pci address) to the fixed interface name - e.g. "eth0",

```

       Example 4. /etc/systemd/network/10-internet.link

       This example assigns the fixed name "internet0" to the interface
       with the device path "pci-0000:00:1a.0-*":

           [Match]
           Path=pci-0000:00:1a.0-*

           [Link]
           Name=internet0

       Example 5. /etc/systemd/network/25-wireless.link

       Here's an overly complex example that shows the use of a large
       number of [Match] and [Link] settings.

           [Match]
           MACAddress=12:34:56:78:9a:bc
           Driver=brcmsmac
           Path=pci-0000:02:00.0-*
           Type=wlan
           Virtualization=no
           Host=my-laptop
           Architecture=x86-64

           [Link]
           Name=wireless0
           MTUBytes=1450
           BitsPerSecond=10M
           WakeOnLan=magic
           MACAddress=cb:a9:87:65:43:21
```


reference: 
https://man7.org/linux/man-pages/man5/systemd.link.5.html